src/lib/GenericPredictor.cc

```diff
@@ -36,7 +36,7 @@ BranchPrediction GenericPredictor::predict(uint64_t address, BranchType type,
   bool direction =
       btb_[hashedIndex].first < (1 << (satCntBits_ - 1)) ? false : true;
   uint64_t target =
-      (knownTarget != 0) ? address + knownTarget : btb_[hashedIndex].second;
+      (knownOffset != 0) ? address + knownOffset : btb_[hashedIndex].second;
```

Comment:
If `knownOffset` is a negative value, will the target be calculated correctly? i.e. is `address + knownOffset` calculated as signed or unsigned before casting implicitly to uint64_t

Reply:
If result of the addition is +ve the result is as expected even if offset is a negative value. e.g. address = 4000, offset = -100, target will be 3900. If the result is negative then this will underflow leading to target being a huge value, but target should never be negative anyway